### DISCUSSION TITLE
Can i use action as a full job to encapsulate all CI process?

### DISCUSSION THREAD

**Discussion by sergeevNick** [community/community#25072] [Actions]
<p>What i want to reach is encapsulate all CI process in one action.<br>
I transform every step into JS action. There is no problem.<br>
How can i use these JS actions in my action that combines them together?</p>
<p>What can i do now</p>
<pre><code>name: Index

on: [push]

jobs:
  build:
    runs-on: ubuntu-latest
    container:
      image: node:lts-alpine

    steps:
      - uses: owner/install@master
      - uses: owner/test@master
      - uses: owner/build@master
      - uses: owner/deploy@master
</code></pre>
<p>But how turn it into something like this?</p>
<pre><code>name: Index

on: [push]

uses: service-pipeline@master
</code></pre>
<p>The main reason is reduce duplicated code that needs to be maintained.</p>
<p>For example, Jenkins has shared libraries that allows to do such thing.</p>
<p>Does GitHub Actions support such type of encapsulation?</p>
<p>Thanks!</p>

**Comment by BrightRan** [comment, marked as answer]
<p>Your request is quite similar with job/step templates, just like the Templates feature for YAML pipelines on Azure Pipelines.<br>
I tested and tried, unfortunately, GitHub Actions seems does not support job/step templates. And it also is very hard to encapsulate multiple jobs/steps into one action.<br>
Looks like, currently we have no available methods to encapsulate multiple jobs/steps into one action, or setup job/step templates.</p>

**Comment by sfynx** [comment]
<p>Would you know if such a feature is planned? This thread indicates there is an open feedback issue for this:</p>
<p><a href="https://github.community/t/being-dry-centralized-workflows/16548">https://github.community/t5/GitHub-Actions/Being-DRY-centralized-workflows/td-p/34485</a></p>
<p>If we cannot service many projects with the same pipeline logic like we can with Jenkins shared libraries, it is impossible to switch to GitHub Actions without creating a large maintenance burden on the pipelines themselves. Some provision for keeping the configurations in the project repositories as small as possible is needed for that.</p>

**Comment by sadiya000000547** [comment]
I don't know what you're trying to tell me. Tell me if my account is 100% safe.If there is a problem with my account, can I disable my account?

**Comment by sadiya000000547** [comment]
Can you explain how my Gmail suddenly disappeared?
